src/Kunstmaan/MediaBundle/Helper/MediaManager.php

```diff
@@ -160,4 +184,26 @@ public function getFolderAddActions()
 
         return $result;
     }
+
+    public function isExtensionAllowed(File $file): bool
+    {
+        if (!$this->limitAllowedExtesions) {
+            return true;
+        }
+
+        $mimeType = $this->mimeTypes->guessMimeType($file->getRealPath());
+        $extensions = $this->mimeTypes->getExtensions($mimeType);
+        $extension = array_shift($extensions);
+        if (!in_array($extension, $this->allowedExtensions, true)) {
+            return false;
+        }
+
+        if (!in_array($extension, $this->imageExtensions, true)) {
+            return true;
+        }
+
+        $errors = $this->validator->validate($file, new Image(['detectCorrupted' => true]));
+
+        return 0 === $errors->count();
```

Comment:
@acrobat zou het voor jouw ok zijn als we de translation gewoon wat meer inclusive maken? Bv 

"Your file has been rejected by the server because either the extension is not allowed or the file is corrupt"

?

Reply:
Imo we shouldn't group exception messages as it's not clear to the user what the exact problem is and how they can resolve it.